database/models.go

```diff
@@ -71,6 +65,7 @@ type Layer struct {
 type Namespace struct {
 	Name          string
 	VersionFormat string
+	DetectedBy    Detector
```

Comment:
Using Detector here maybe not valid. As the reason shows in the database, namespace/feature may be detected by two different detectors in two different layers.